processing/src/main/java/org/apache/druid/segment/AutoTypeColumnIndexer.java

```diff
@@ -459,11 +461,12 @@ public Object getObject()
         if (0 <= dimIndex && dimIndex < dims.length) {
           final StructuredData data = (StructuredData) dims[dimIndex];
           if (data != null) {
-            return ExprEval.bestEffortOf(data.getValue()).valueOrDefault();
+            final Object o = ExprEval.bestEffortOf(data.getValue()).valueOrDefault();
+            return o == null ? defaultValue : o;
```

Comment:
I did this to make things behave consistently for realtime selectors of 'auto' typed columns as when those columns are persisted, rather than to pre-empitively eliminate null values from the indexer like the regular numeric indexers do in default value mode. So its not at all for callers, and just for consistency in default value mode when we are dealing with a single type numeric.